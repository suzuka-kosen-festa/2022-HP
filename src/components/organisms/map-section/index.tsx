import React from "react";
import tw from "twin.macro";
import type { MapSectionProperties } from "../../../models";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";
import { Map } from "../../molecules/map";

const MapSectionContainer = tw.section`flex flex-col space-y-[calc(200vw / 63)]`;

const MapBox = tw.div`flex flex-col space-y-[calc(200vw / 63)] lg:flex-row-reverse lg:space-x-[calc(800vw / 189)] lg:space-y-0 lg:space-x-reverse justify-end`;

const MapSection: React.FC<MapSectionProperties> = ({ center, children, label, title, zoom, ...rest }) => (
  <MapSectionContainer aria-label={"アクセス"} {...rest}>
    <Heading>{title}</Heading>
    <MapBox>
      <Text css={tw`whitespace-pre-wrap select-text`}>&#12306;{children}</Text>
      <Map center={center} title={label} zoom={zoom} />
    </MapBox>
  </MapSectionContainer>
);

export { MapSection };
