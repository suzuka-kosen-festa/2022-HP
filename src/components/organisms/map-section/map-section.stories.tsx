import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { MapSection } from ".";

type T = typeof MapSection;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  title: "Map",
};

export default {
  args,
  argTypes: {
    title: {
      description: "Title of section",
      control: {
        type: "text",
      },
    },
  },
  component: MapSection,
} as Meta;

export const Default: Story = {};
