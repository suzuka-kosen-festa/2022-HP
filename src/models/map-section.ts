import type { ComponentProps, ReactHTML } from "react";
import type { MapProperties } from ".";

export type MapSectionProperties = Omit<ComponentProps<ReactHTML["section"]>, "children"> &
  Omit<MapProperties, "title" | "children"> & {
    title: string;
    label: string;
    children: string;
  };
