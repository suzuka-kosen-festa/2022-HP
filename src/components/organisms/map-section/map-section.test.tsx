import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./map-section.stories"

const { Default } = composeStories(stories);

const regionoptions = {
  name: "アクセス"
}

describe("(components) organisms/map-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />)
    expect(getByRole("region", regionoptions))
  })
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
