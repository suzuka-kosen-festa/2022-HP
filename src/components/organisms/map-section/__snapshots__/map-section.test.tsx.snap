// Vitest Snapshot v1

exports[`(components) organisms/map-section > take snap shot 1`] = `
<div>
  <section
    aria-label="map"
    class="css-wkpw2c"
  >
    <h1
      class="css-4pmhej-MapSection"
    >
      Map
    </h1>
    <a
      class="css-1l4w6pd"
      href="https://res.cloudinary.com/dvgtaafq5/image/upload/q_60/f_auto/v1665765554/map_vjyaf2.webp"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="校内マップ"
        class="css-1h35ydv-MapSection"
        height="2915"
        src="/statics/map.webp"
        width="4128"
      />
    </a>
  </section>
</div>
`;
